### PR TITLE
ISPN-9083 remove deprecated query annotations and deprecated index by default functionality

### PR DESCRIPTION
**Pull request by anistor**
https://issues.jboss.org/browse/ISPN-9083
https://issues.jboss.org/browse/ISPN-9322
https://issues.jboss.org/browse/ISPN-6161

### CODE REVIEW

**Comment by anistor** [comment]
Rebased.

**Comment by tristantarrant** [comment]
Quite a few REST search failures probably related to the changes here.

**Comment by gustavonalle** [comment]
Still failures in REST, but it seems CI didn't run with the last commit, re-triggering...

**Comment by anistor** [comment]
Rebased. And REST failures should be fixed now.

**Comment by anistor** [comment]
Fixed last failing REST test : CacheV2ResourceTest

File: remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/logging/Log.java [hunk 1]
```diff
@@ -52,12 +49,12 @@
...
-         " annotations defined in file '%s'. Please annotate your message type for indexing or add" +
-         " 'option indexed_by_default = false;' to your schema file to disable indexing of message types" +
-         " that do not have indexing annotations.", id = 28012)
-   void legacyIndexingIsDeprecated(String typeName, String fileName);
+//   @LogMessage(level = WARN)
+//   @Message(value = "Using deprecated legacy indexing mode (all fields) for message type '%s' missing indexing" +
+//         " annotations defined in file '%s'. Please annotate your message type for indexing or add" +
+//         " 'option indexed_by_default = false;' to your schema file to disable indexing of message types" +
+//         " that do not have indexing annotations.", id = 28012)
+//   void legacyIndexingIsDeprecated(String typeName, String fileName);
```
Comment (gustavonalle) [inline remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/logging/Log.java]: Remove?

Comment (anistor) [reply remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/logging/Log.java]: sure. but I think we had the rather strange rule of not removing unused log messages, just commenting them out. not sure why. I don't give a :rat:'s :ass: about them. will remove!

Comment (ryanemerson) [reply remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/logging/Log.java]: Commenting them is better. That way we can't accidentally re-use an old ID.

**Comment by anistor** [comment]
Too late. I removed it. :) We will not reuse them because we are supposed to only add at the end of the range.

**Pull request merged**